CMakeLists.txt

```
@@ -1089,8 +1089,16 @@ mark_as_advanced(MAXIMIZE_CPU_USAGE)
 # Doxygen
 if(BUILD_DOCUMENTATION)
   # need doxygen
-  find_package(Doxygen REQUIRED)
+  find_package(Doxygen 1.8.7 REQUIRED)
 
+  if(NOT Doxygen_FOUND)
+    find_package(Doxygen 1.8.8 REQUIRED)
+  endif()
+
+  if(NOT Doxygen_FOUND)
+    message(FATAL_ERROR "Doxygen version 1.8.7 or 1.8.8 not found")
+  endif()
+  
```

If REQUIRED, it'll throw directly, so no need for NOT Doxygen_FOUND. Also, cmake find_package has a version range available.

```suggestion
  find_package(Doxygen 1.8.7...<1.8.9 REQUIRED)
```

---

cf https://cmake.org/cmake/help/latest/command/find_package.html

---

> By specifying <, the upper end point will be excluded. Version ranges are only supported with CMake 3.19 or later

We request 3.20 already, so it's ok.